### PR TITLE
Spellcasting element

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request 
Remake of #8901, but with elements and no extras. Should be fine.

## Why It's Good For The Game
Less hardcoded spellcasting requirements, specially clothes ones.

## Changelog
:cl:
refactor: refactored spell casting requirements.
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/wiz_robe.dm [hunk 1]
```diff
@@ -117,29 +129,20 @@
 	item_state = "santa"
 
 /obj/item/clothing/suit/wizrobe/fake
-	name = "wizard robe"
```
Comment (Putnam3145) [inline code/modules/clothing/suits/wiz_robe.dm:120]: Looks like you removed some copypasta here, which is obvs good, but the fake wizard robes inherit the magic_flags, as far as I can tell?

Comment (Putnam3145) [reply code/modules/clothing/suits/wiz_robe.dm:120]: oh and the same stuff goes for all the other fake robes obvs, i could be misunderstanding something here

File: code/modules/mob/living/simple_animal/friendly/gondola.dm [hunk 2]
```diff
@@ -33,6 +33,10 @@
 	if (!(istype(src, /mob/living/simple_animal/pet/gondola/gondolapod)))
 		CreateGondola()
 
+/mob/living/simple_animal/pet/gondola/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/spellcasting, SPELL_SKIP_VOCAL) // so they can cast spells despite being silent.
```
Comment (Putnam3145) [inline code/modules/mob/living/simple_animal/friendly/gondola.dm:38]: fear

**Review by Putnam3145 — COMMENTED** [review]
Final thought: I know it's all being fixed, but the inconsistency with 0/1 FALSE/TRUE is killing me in the previous code

File: code/modules/clothing/suits/wiz_robe.dm [hunk 3]
```diff
@@ -41,7 +42,7 @@
 	permeability_coefficient = 1
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
 	resistance_flags = FLAMMABLE
-	dog_fashion = /datum/dog_fashion/head/blue_wizard
```
Comment (Putnam3145) [inline code/modules/clothing/suits/wiz_robe.dm:39]: removing ian's wizard hat?

Comment (Ghommie) [reply code/modules/clothing/suits/wiz_robe.dm:39]: It's the same value of `/obj/item/clothing/head/wizard/var/dog_fashion`

**Review by silicons — APPROVED** [review]

**Pull request merged**